server/src/main/java/io/crate/execution/ddl/tables/MappingUtil.java

```diff
@@ -0,0 +1,210 @@
+        for (Reference child: children) {
+            allColumnsMap.put(child.column().leafName(), addColumnProperties(child, tree));
+        }
+        return allColumnsMap;
+    }
+
+    /**
+     * Aligned with AnalyzedColumnDefinition.toMapping()
+     */
+    private static Map<String, Object> addColumnProperties(Reference reference, HashMap<ColumnIdent, List<Reference>> tree) {
+
+        Map<String, Object> columnProperties = reference.toMapping();
+        if (reference.valueType().id() == ArrayType.ID) {
+            HashMap<String, Object> outerMapping = new HashMap<>();
+            outerMapping.put("type", "array");
+            if (ArrayType.unnest(reference.valueType()).id() == ObjectType.ID) {
+                objectMapping(columnProperties, reference, tree);
+            }
+            outerMapping.put("inner", columnProperties);
+            return outerMapping;
+        } else if (reference.valueType().id() == ObjectType.ID) {
+            objectMapping(columnProperties, reference, tree);
+        }
+        return columnProperties;
+    }
+
+    private static void objectMapping(Map<String, Object> propertiesMap, Reference reference, HashMap<ColumnIdent, List<Reference>> tree) {
+        propertiesMap.put("dynamic", ColumnPolicies.encodeMappingValue(reference.columnPolicy()));
+        Map<String, Map<String, Object>> nestedObjectMap = createPropertiesMap(reference.column(), tree);
+        if (nestedObjectMap != null) {
+            propertiesMap.put("properties", nestedObjectMap);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public static void mergeConstraints(Map<String, Object> meta,
+                                         List<Reference> references,
+                                         IntArrayList pKeyIndices,
+                                         Map<String, String> checkConstraints) {
+
+        // CHECK
+        if (checkConstraints.isEmpty() == false) {
+            Map<String, String> existingCheckConstraints = (Map<String, String>) meta.get("check_constraints");
+            if (existingCheckConstraints == null) {
+                existingCheckConstraints = new HashMap<>();
+                meta.put("check_constraints", existingCheckConstraints);
+            }
+            for (var entry : checkConstraints.entrySet()) {
+                String name = entry.getKey();
+                String expression = entry.getValue();
+                existingCheckConstraints.put(name, expression);
+            }
+        }
+
+        // PK
+        if (pKeyIndices.isEmpty() == false) {
+            List<String> primaryKeys = (List<String>) meta.get("primary_keys");
+            if (primaryKeys == null) {
+                primaryKeys = new ArrayList<>();
+                meta.put("primary_keys", primaryKeys);
+            }
+            for (int i = 0; i < pKeyIndices.size(); i ++) {
+                primaryKeys.add(references.get(pKeyIndices.get(i)).column().fqn());
+            }
```

Comment:
It was bug on master, merged https://github.com/crate/crate/commit/1963ea9a499afb724871fabe2922db1017dc8e78